521 assure users that they download the zeeguu reader from a credible source

igawaclawska opened the pull request:
_This PR does not include illustration updates. Once the illustration is done (probably in week 40), it can be added as a new PR, or as an update to this one._ 

# Major visual updates:

## `SupportedNotification_NotInstalled.js` modal

| Before redesign  | 
|-----------|
| <img width="400px" src="https://github.com/user-attachments/assets/85cad908-f64d-4971-84d6-211c2fb07e08" alt="Screenshot 2024-09-26 at 13:19:25"> | 

| New for Mozilla  | New for Chrome and supported  |
|-----------|-----------|
| <img width="500px" src="https://github.com/user-attachments/assets/5e0cf4a7-26a0-40f6-a84d-98a2c4a0835b" alt="Screenshot 2024-09-26 at 13 18 57">  | <img width="500px" src="https://github.com/user-attachments/assets/5b44a17f-c65e-4f0c-9abb-ab67926b84cb" alt="Screenshot 2024-09-26 at 13 18 29"> |


## `InstallExtension.js` page
| Before redesign  | 
|-----------|
| <img width="400px" src="https://github.com/user-attachments/assets/c3ec8713-64ff-48cb-9a62-30b569b72a36" alt="Screenshot 2024-09-26 at 13 24 01"> | 

| New for Mozilla  | New for Chrome and supported  |
|-----------|-----------|
| <img width="500px" src="https://github.com/user-attachments/assets/00dd79bb-8587-4755-beb1-4125fbc702bd" alt="Screenshot 2024-09-26 at 13 23 16"> | <img width="500px" src="https://github.com/user-attachments/assets/440979e9-39a2-48c9-a4e7-1834876e9f55" alt="Screenshot 2024-09-26 at 13 22 26"> |

## `ExtensionMessage.js` pop-up is now removed (as discussed in the issue: https://github.com/zeeguu/web/issues/521 )



## Code updates:

- The `GoToButton` and `WarningButton` components from `modal_shared` have been replaced with the `Button` component from the `_pages_shared` folder. The `Button` component from the `_pages_shared` folder will be moved to the `components` folder during further refactoring. Also, please note that I have not used the legacy button components from the `AllButtons` file. It's because these buttons are made from div HTML elements instead of button elements, which makes them not keyboard-focusable.   
- Same as preference pages, modals now use the `redirect()` function for navigation, which is applied to the onClick event handler of the Button instead of using link wrappers and a href attribute.
- the `redirect` function has now been enhanced to let developers decide whether links should be opened in the same or a new window
```
export default function redirect(linkToRedirect, openInNewWindow = false) {
  if (openInNewWindow) {
    window.open(linkToRedirect, '_blank', 'noopener,noreferrer');
  } else {
    window.location.href = linkToRedirect;
  }
}
```
- Replaced **simple** components like `Header.js` with `Header.sc.js` where it was possible (for now, in the folders belonging to the modals, I will do the same for the Preference Pages)

tfnribeiro commented on src/components/redirect_notification/SupportedNotification_NotInstalled.js:
```diff
@@ -22,26 +24,39 @@ export default function SupportedNotification_NotInstalled({
-        <p>
-          To read this article with the help of Zeeguu without the extension,
-          simply click "Add to Saves" above the article's title.
+        <FullWidthImage
```
I can see that with the Header you have dropped it as a component an instead you import the styling which you then pass the props as normal. I can see that the `FullWidthImage` is essentially the same. Wouldn't it be better to just import the styling and not be dependent on the having the props setup in the component? 

I think this was similar to what happened with the `Button` Component which didn't have a Prop for type and then it didn't work when I was trying to assign the `type="submit"`, and if it was just a styled component we wouldn't have this problem. I see you have also deleted the `Button`. So what's the pattern here?

tfnribeiro commented on src/utils/routing/routing.js:
Good doc string!

tfnribeiro reviewed (commented):
Overall, really good work!

However, I don't understand why in some cases we deleted the Wrapper components with styling and in others we have kept them. Mainly the FullWidthImage VS Button for example.

I think it seems like a better pattern of just using the Styled, rather than creating a Component which doesn't really have any other components in itself. To me, the logic would be: If you are just styling one HTML tag essentially, then that should be a StyledComponent, if it's a composition of a few components. let's say a button with a specific Icon that opens a modal, in that case, we use a React Component, since they need some Logic. Let me know what is the logic you are going for - maybe I am missing something.

igawaclawska commented:
> However, I don't understand why in some cases we deleted the Wrapper components with styling and in others we have kept them. Mainly the FullWidthImage VS Button for example.

I'm going to follow **the Styled components pattern** for simple components that only take children. However, for now, I changed the pattern only for components from the modal_shared (except the button, which originally comes from the `_pages _shared` folder). I'm going to continue with the rest when working on the next issue (https://github.com/zeeguu/web/issues/551)

tfnribeiro commented:
> > However, I don't understand why in some cases we deleted the Wrapper components with styling and in others we have kept them. Mainly the FullWidthImage VS Button for example.
> 
> I'm going to follow **the Styled components pattern** for simple components that only take children. However, for now, I changed the pattern only for components from the `modal_shared` folder(except the button, which originally comes from the `_pages _shared` folder). I'm going to continue with the rest when working on the next issue (#551)

I read now the last line of the PR and I can see what you said there, which seems to be what I was trying to define :) 

> Replaced simple components like Header.js with Header.sc.js where it was possible (for now, in the folders belonging to the modals, I will do the same for the Preference Pages)

tfnribeiro commented:
> If you prefer, I can also add these additional component changes to this PR. I just didn't want to make as huge code modification all at once :) . Otherwise - I will continue refactoring when working on this issue (#551), as initially planned.

It's all good - I was just trying to understand what was the pattern we are going for!

mircealungu commented on src/articles/ExtensionMessage.js:
```diff
@@ -43,27 +47,40 @@ export default function ExtensionMessage({
           <Heading>
-            <span className="annotation">New!</span>&nbsp;
-            {strings.extensionHeadline}
+            <span className="annotation">New!</span>&nbsp;Read and translate
```
I think we agreed in another discussion that this message with `New!` is going away? It's not "new" anymore anyway :)

igawaclawska replied:
Done  ✅  But this pop-up will be deleted soon :)

mircealungu reviewed (approved):
looks good.
minor ask and we're merging this in!

mircealungu commented:
I spent too much time reading all the text, and then lost the connection between the title and the button. 

What about a simpler modal? 

<img width="716" alt="image" src="https://github.com/user-attachments/assets/1bbb764b-3d47-4b15-823e-235244bc4f88">

"*" means that there's an anchor somewhere that points to it. something like this: 

<img width="721" alt="image" src="https://github.com/user-attachments/assets/51500fe2-3997-4d04-bf9f-cf27f3c56920">

or like this: 

<img width="722" alt="image" src="https://github.com/user-attachments/assets/15a7c157-21d2-449f-837b-90c6124ed954">

or, if we don't like the * in the button text, we can clarify things with a bit more text

<img width="774" alt="image" src="https://github.com/user-attachments/assets/589e9cc1-7c4b-4b7f-b4b7-af81da0c9bf0">


Note that i've removed the Save text, again for simplifying the dialog... On the mobile phone it matters most, but for that we use another modal, right? I know you asked about it Iga, and I don't remember what I said, but if you ask me now, i'd remove it... 

Finally, did we not learn while testing with Sandro that it works also on Arc? Maybe add Arc there too? 

A final iteration could allow the possibility of other browsers like this (though we should not worry too much about people who use such obscure browsers ... they're probably less than 1% and as soon as a new browser becomes popular we can simply add it here)

<img width="772" alt="image" src="https://github.com/user-attachments/assets/dcda905e-1432-455c-8ede-6f9b3c7ef60b">

igawaclawska commented on src/articles/ExtensionMessage.js:
```diff
@@ -1,18 +1,22 @@
+  isMobile,
+  runningInChromeDesktop,
+} from "../utils/misc/browserDetection";
+import ArrowForwardRoundedIcon from "@mui/icons-material/ArrowForwardRounded";
```
I agree. However, this name is defined by the creators of MUI icons, and if I change it, the import link will not match.

mircealungu replied:
ah, they should read https://writings.stephenwolfram.com/2010/10/the-poetry-of-function-naming/ by Wolfram :)

And in this case, we're lucky. Because it's a default import you can name it whatever you like ;)

mircealungu replied:
so `import RoundedForwardArrow from "@mui/icons-material/ArrowForwardRounded";` should work like a breeze.

igawaclawska replied:
✅ Done

igawaclawska commented on src/articles/ExtensionMessage.js:
```diff
@@ -1,18 +1,22 @@
 import LocalStorage from "../assorted/LocalStorage";
-import strings from "../i18n/definitions";
+import redirect from "../utils/routing/routing";
+import { getExtensionInstallationButtonContent } from "../utils/extension/extensionInstallationButtonContent";
```
As we agreed here (https://github.com/zeeguu/web/issues/521) - this `ExtensionMessage.js` pop-up from `/articles` is to be deleted (probably in the next PR).

The `getExtensionInstallationLinks` and `getExtensionInstallationButtonContent` functions are **shared** and used by the **modal** - `SupportedNotification_NotInstalled.js` from `components/redirect_notification`, and the **onboarding page** - `InstallExtension.js` from `pages/onboarding` so I personally think they shouldn't be moved to the `/articles` folder. I propose that we could still keep them where they are: `utils/extension`, but they could be grouped in their own folder, called for example `extension_installation`

igawaclawska replied:
Optionally, in the future (next PRs), I would also see the `getExtensionInstallationLinks` and `getExtensionInstallationButtonContent` merged into one function that returns an object containing the name of the browser store and links to it :) But first, I would need to check how intuitive it looks in practice.

mircealungu replied:
ok, it's more complicated than i thought :)
could we then move everything related to extension installation not in utils, but somewhere in /components? 
i just have this feeling of what's in `/utils` is something we didn't think enough about :)

igawaclawska replied:
> ok, it's more complicated than i thought :)

Yes :) 

> could we then move everything related to extension installation not in utils, but somewhere in /components?
i just have this feeling of what's in /utils is something we didn't think enough about :)

We could move them to `/components`, but I'm unsure how we could allocate them there. These functions have very specific purpose, and are not components. If we group them in a folder, we need to name the folder very well so that it is clear that these functions represent shared behavior, instead of components.

igawaclawska commented:
For consistency, I've updated the content of both: `SupportedNotification_NotInstalled.js` modal and `InstallExtension.js` page. 

For visual balance, I have decided to keep the annotation above the install button on the `InstallExtension.js` page (because of the `I will install it later` button)

<img width="688" alt="Screenshot 2024-09-27 at 10 59 33" src="https://github.com/user-attachments/assets/f79acc08-dcbc-48e4-9634-e85b4c03cb9c">
<img width="960" alt="Screenshot 2024-09-27 at 10 59 12" src="https://github.com/user-attachments/assets/0e56cee2-3055-4e55-acb0-f4d484286f18">

mircealungu commented:
looks good!

igawaclawska commented:
Latest update: The `ExtensionMessage.js` pop-up with all its dependencies is now removed, as discussed in this issue (https://github.com/zeeguu/web/issues/521)

mircealungu commented:
nice! let's deploy the new cool dialogs :) happy new weekend!

igawaclawska commented:
>nice! let's deploy the new cool dialogs :) happy new weekend!

Cool! Have a great weekend, too :)

mircealungu commented:
I think it's time to organize three next usability test :)

M.

On Sat, 28 Sep 2024 at 10.36, Iga ***@***.***> wrote:

> nice! let's deploy the new cool dialogs :) happy new weekend!
>
> Cool! Have a great weekend, too :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zeeguu/web/pull/560#issuecomment-2380552104>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AADRNB2AHTG6XUFQ4A423TDZYZL67AVCNFSM6AAAAABO43QGPWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGOBQGU2TEMJQGQ>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
